tests/src/test/scala/org/http4s/UriSpec.scala

```diff
@@ -57,14 +57,23 @@ class UriSpec extends Http4sSpec with MustThrownMatchers {
       }
 
       "parse port correctly" >> {
-        "if there is one" in {
-          val uri = getUri("http://localhost:8080/")
-          uri.port must_=== Some(8080)
-        }
+        "if there is a valid (non-negative) one" >> prop { nonNegative: Int =>
+          {
+            val uri = getUri(s"http://localhost:$nonNegative/")
+            uri.port must_=== Some(nonNegative)
+          }
+        }.setGen(Gen.choose[Int](0, Int.MaxValue))
         "if there is none" in {
           val uri = getUri("http://localhost/")
           uri.port must_=== None
         }
+        // See RFC 3986, section 6.2.3
+        "for an empty String" in {
```

Comment:
Given `https://foobar:/` should this infer `443` rather than `80`? I worry as I believe some may rely on this behavior by accident and then they would go to the unsecured port by default.

Reply:
Are we inferring anywhere in the Uri?  We do in the clients, but the Uri has an optional port.  You're correct that the inference is scheme specific, but I think this test is right.

Reply:
> Given https://foobar:/ should this infer 443 rather than 80?

Is it worthwhile to open another issue, @ChristopherDavenport, if `https://foobar:/` resolves a port of 80 rather than 443? I'm only asking since, I believe, my PR addresses https://github.com/http4s/http4s/issues/2583.

Reply:
Hi @ChristopherDavenport -

> Given https://foobar:/ should this infer 443 rather than 80?

So the following looks right or wrong to you? I admit to not knowing how the `Port` is used if there's a supplied `Scheme`, i.e. my first `Uri#fromString` REPL example.

```scala
@ import $ivy.`org.http4s::http4s-core:0.20.6` 
import $ivy.$                               

@ Uri.fromString("https://www.google.com:80") 
res5: org.http4s.package.ParseResult[Uri] = Right(Uri(Some(Scheme(https)), Some(Authority(None, RegName(www.google.com), Some(80))), "", , None))

@ Uri.fromString("https://www.google.com:/") 
res6: org.http4s.package.ParseResult[Uri] = Right(Uri(Some(Scheme(https)), Some(Authority(None, RegName(www.google.com), None)), "/", , None))
```

Does this look right, Chris?